docs/dev-environment-setup.md

```diff
@@ -0,0 +1,233 @@
+
+## 6. Deploy the Development Database
+
+```
+createdb ciip_portal_dev ciip_portal_test
+```
+
+`psql --list` to verify the databases were created.
+
+Rather than interacting with Sqitch directly, we use a [custom script](https://github.com/bcgov/cas-ciip-portal/blob/develop/schema/data/deploy-data.sh) for convenience to deploy the database, including seed data for dev, test and production.
+
+
+From the project root, run `.bin/deploy-data.sh --help` to see possible options. In development, this is most commonly run with `.bin/deploy-data.sh -d -dev` to drop the current database and redeploy after pulling down database-changing work from the remote or tampering with the database.
+
+## 7. Install and Run the Web Application
+
+Install the `node_modules` including `devDependencies` from the `package.json`:
+```
+cd app && yarn --dev
+```
+
+Run the development server with `yarn dev` and navigate in your browser to `localhost:3004`.
+
+Work that changes Relay GraphQL queries will require you to re-run `yarn build:relay` (which alternatively can be run using a `--watch` flag).
+
+## 7.5 Bypass Authentication in Development
+
+There are some shortcuts to bypass authentication in development and run the server while logged in as a preset user:
+
+```
+cd app && yarn dev <AS_USER>
+```
+...where `<AS_USER>` is one of the following:
+
+```
+AS_REPORTER
+AS_CERTIFIER
+AS_ANALYST
+AS_ADMIN
+```
+
+[Learn more here](https://github.com/bcgov/cas-ciip-portal/blob/develop/docs/all-views-inventory.md) about the functionality available to each user type.
+
+## 8. Run Tests
+
+### Unit Tests with Jest
+
+```
+cd app && yarn test
+```
+Front-end unit tests are heavily snapshot-based. Work that changes the DOM will result in a diff from the last accepted snapshot and cause related tests to fail. You can update the snapshots and review / accept the diff with `yarn test -u`.
+
+### End-to-end Tests with Cypress
+
+```
+cd app && yarn cypress
+```
+
+### Database Unit Tests with pgTAP
+
+```
+cd schema && make unit
+```
+
+## 9. Tidy Your Work
+
+[pre-commit](https://pre-commit.com/) runs a variety of formatting and lint checks configured in [`.pre-commit-config.yaml`](https://github.com/bcgov/cas-ciip-portal/blob/develop/.pre-commit-config.yaml) which are required for a pull request to pass CircleCI.
+
+`pre-commit install` will [configure a pre-commit hook to run before every commit](https://pre-commit.com/#usage); alternatively, you can run it manually with:
+
+```
+pre-commit run --all-files
+```
+
+If you are impatient and your work is isolated to Javascript, it may be faster to run only the related formatter (`xo`), but it may not catch everything (such as the end-of-file fixer and trailing whitespace):
+
+```
+yarn format
+```
+
+<!-- ### Start & configure psql server
+- https://www.postgresql.org/download/macosx/ for an installer
```

Comment:
I've stuck with guiding the user to try to let asdf install Postgres (with XML plugin option) - given nothing else works, I presume they would be able to find the manual download on their own?

Reply:
Probably, there's a million ways of installing psql on OSX through the standalone app, or homebrew, or many others.
Since asdf is installing it for that particular version and application, does it make sense to use it vs having a single local postgres server for all apps?

Reply:
> Since asdf is installing it for that particular version and application, does it make sense to use it vs having a single local postgres server for all apps?

One reason for using asdf is to ensure they install the specific version of Postgres - you'll notice in the `.tool-versions` we specify 11.4 (whereas Postgres is now up to 13).

I guess the other idea behind using it is that a developer may use their machine for other applications, for which they may use other package managers and other database versions.

That said, there's a step which [sets the asdf installed version of postgres as global](https://github.com/bcgov/cas-ciip-portal/pull/1133/files#diff-fb7bdbe3eb6696759bbe84ebb71578a5R25), for Sqitch to recognize the existing Postgres when installed instead of installing its own.

Reply:
We should update our tool versions to use postgres 12.2, as it is the version that is used in OpenShift

Reply:
Re - sqitch and having postgres set as global, is that only required for installation, or for running sqitch as well?

Reply:
I'm not entirely sure. It comes up because without doing so, Sqitch did not find the right Postgres and instead installed another copy of Postgres during its own installation. I think this also tells it which `psql` to use during runtime.

I installed via asdf, but there's a bit of description about this on the [Sqitch Homebrew formula](https://github.com/sqitchers/homebrew-sqitch#--with-postgres-support).

Reply:
(also, new ticket for updating Postgres to 12: https://youtrack.button.is/issue/GGIRCS-2003)